src/inject/inject.js

```diff
@@ -9,16 +9,36 @@ const NAME = "test";
 const addElement = (e) => {
   acknowledge(e);
 
+  const target = e.target.querySelector(`[data-name="${NAME}"]`);
+
+  // Events are received that are intended for all consumers. 
+  // This makes sure the extension only handles intended for it.
+  if(!target) {
+    return;
+  }
+
+  // This container has already been handled by this extension before.
+  // To avoid duplicates it will be ignored. 
+  if(target.classList.contains(`${NAME}-handled`)) {
+    return;
+  }
+
+  // Add a class so the extension on refire can tell this target has been handled before.
+  target.classList.add(`${NAME}-handled`);
```

Comment:
This should go inside the `requestAnimationFrame` call too, since changing class names can cause repaints.

Reply:
... the `if (handled) return` should probably go inside the raf too.  Alternative to className: 

```
target[`extension-${NAME}`] = true
```

Reply:
If handled, the title should probably be updated with the new data.